techlibs/quicklogic/synth_quicklogic.cc

```diff
@@ -0,0 +1,260 @@
+                continue;
+            }
+            if (args[argidx] == "-no_abc_opt") {
+                abcOpt = false;
+                continue;
+            }
+            
+            break;
+        }
+        extra_args(args, argidx, design);
+
+        if (!design->full_selection())
+            log_cmd_error("This command only operates on fully selected designs!\n");
+
+        log_header(design, "Executing SYNTH_QUICKLOGIC pass.\n");
+        log_push();
+
+        run_script(design, run_from, run_to);
+
+        log_pop();
+    }
+
+    void script() override
+    {
+        if (check_label("begin")) {
+            std::string readVelArgs;
+            readVelArgs = " +/quicklogic/" + family + "_cells_sim.v";
+            
+            run("read_verilog -lib -specify +/quicklogic/cells_sim.v" + readVelArgs);
+            run(stringf("hierarchy -check %s", help_mode ? "-top <top>" : top_opt.c_str()));
+        }
+
+        if (check_label("prepare"))  {
+            run("proc");
+            run("flatten");
+            run("opt_expr");
+            run("opt_clean");
+            run("deminout");
+            run("opt");
+        }
+
+        if (check_label("coarse")) {
+            run("opt_expr");
+            run("opt_clean");
+            run("check");
+            run("opt");
+            run("wreduce -keepdc");
+            run("peepopt");
+            run("pmuxtree");
+            run("opt_clean");
+
+            run("alumacc");
+            run("opt");
+            run("fsm");
+            run("opt -fast");
+            run("memory -nomap");
+            run("opt_clean");
+        }
+
+        if (check_label("map_ffram")) {
+            run("opt -fast -mux_undef -undriven -fine");
+            run("memory_map -iattr -attr !ram_block -attr !rom_block -attr logic_block "
+                    "-attr syn_ramstyle=auto -attr syn_ramstyle=registers "
+                    "-attr syn_romstyle=auto -attr syn_romstyle=logic");
+            run("opt -undriven -fine");
+        }
+
+        if (check_label("map_gates")) {
+            if (inferAdder) {
+                run("techmap -map +/techmap.v -map +/quicklogic/" + family + "_arith_map.v");
+            } else {
+                run("techmap");
+            }
+            run("opt -fast");
+            run("opt_expr");
+            run("opt_merge");
+            run("opt_clean");
+            run("opt");
+        }
+
+        if (check_label("map_ffs")) {
+            
+            run("opt_expr -mux_undef");
+            run("simplemap");
+            run("opt_expr");
+            run("opt_merge");
+            run("opt_clean");
+            run("opt");
+        }
+
+        if (check_label("map_luts")) {
+            run("abc -lut 4 ");
+            run("clean");
+            run("opt_lut");
+        }
+
+        if (check_label("map_cells")) {
+
+            std::string techMapArgs;
+            techMapArgs += " -D NO_LUT -map +/quicklogic/" + family + "_lut_map.v";
```

Comment:
it looks like all the `*_lut_map.v` files are wrapped in: 
```
`ifndef NO_LUT
(...)
`endif
```
this mapping will not do anything. Can we simply remove it?

Reply:
Yes, I removed it.